packages/muelu/src/MueCentral/MueLu_VerboseObject.cpp

```diff
@@ -140,7 +141,7 @@ namespace MueLu {
     int mpiStarted = 0; MPI_Initialized(&mpiStarted);
     if (mpiStarted)     {
       int procRank;
-      MPI_Comm_rank(MPI_COMM_WORLD, &procRank);
+      MPI_Comm_rank(MueLu::GetGlobalComm(), &procRank);
       fn = filename + "." + std::to_string(procRank);
```

Comment:
This doesn't really fix the issue but just moves the `MPI_COMM_WORLD` to a different header.

Reply:
It's unclear if the use of MPI_COMM_WORLD is a problem here.

Reply:
We could change the constructor of VerboseObject to accept a Teuchos::Comm object?

Reply:
Doesn't every single MueLu class derive from `VerboseObject`?

Reply:
- Is `numProcs_` actually used by anything?
- Could we skip the bit with `MPI_COMM_WORLD` and instead set `procRank_` once we have a communicator? We already do that in `CreateXpetraPreconditioner` anyway..

Reply:
`grep` shows that `numProcs_` isn't used anywhere.

> Could we skip the bit with MPI_COMM_WORLD and instead set procRank_ once we have a communicator? We already do that in CreateXpetraPreconditioner anyway..

I *think* this would be fine.

Reply:
Do you want to set the `procRank` in every class that inherits from `BaseClass`? 
If you feel like the use of `MPI_COMM_WORLD` is justified here, we can just add `// CHECK: ALLOW MPI_COMM_WORLD` to those lines.